crates/core/src/operations/load_cdf.rs

```diff
@@ -236,6 +253,11 @@ impl CdfLoadBuilder {
             }
         }
 
+        // All versions were skipped due to date our of range
+        if !self.enable_out_of_range && change_files.is_empty() && add_files.is_empty() && remove_files.is_empty() {
+            return Err(DeltaTableError::ChangeDataTimestampGreaterThanCommit { ending_timestamp: ending_timestamp });
```

Comment:
So according to the link above too, this is only thrown if the timestamp exceeds the last commit timestamp and out of range is not enabled. You should be able to move this up outside the loop to check immediately, I don't think it matters if the actions are empty in this case.